src/search3_app.erl

```diff
@@ -15,15 +16,34 @@
 ]).
 
 start(_StartType, _StartArgs) ->
-    {ok, _} = application:ensure_all_started(grpcbox),
-    Url = config:get("search3", "service_url", "localhost"),
-    Port = case config:get("search3", "service_port", 8443) of
-        8443 -> 8443;
-        Port1 -> list_to_integer(Port1)
-    end,
-    Endpoints = [{http, Url, Port, []}],
-    {ok, _} = grpcbox_channel_sup:start_child(?SEARCH_CHANNEL, Endpoints, #{}),
+    EndPoint = get_endpoints(),
+    application:set_env(search3, service_endpoint, EndPoint),
     search3_sup:start_link().
 
 stop(_State) ->
     ok.
+
+get_endpoints() ->
+    Host = config:get("search3", "service_url", "localhost"),
+    case Host of
+        ":discover" ->
+            discover_endpoints();
+        _ ->
+            Port = config:get_integer("search3", "service_port", 8443),
+            to_conn_str(Host, Port)
```

Comment:
need to change this to: `[to_conn_str(Host, Port)]

Reply:
that doesn't sound right to me, are you sure? I would think `hd(discover_endpoints()).` is more appropriate, so that get_endpoints always returns a 'string'?

Reply:
This should remain a list yeah. What you want likely want to do in search3_rpc.erl is something along the lines of such:

```erlang
get_endpoint() ->
    case get('$search_ep') of
        undefined ->
            EP = select_endpoint(),
            put('$search_ep', EP),
            EP;
        Else ->
            Else
    end.


select_endpoint() ->
    case application:get_env(search3, endpoints) of
        {ok, EndPoints} ->
            lists:nth(rand:uniform(length(EndPoints)), EndPoints);
        _ ->
            erlang:error({search3, no_endpoints_configured})
    end.
```

That assumes that you only have one process in a session thinger at a time. Otherwise you're gonna have to coordinate in some other manner.

Reply:
To clarify: 

1) the `:discover` path returns a list of endpoints
2) the `config:get` path returns just a string
3) My put/get code is only taking into account strings. The change is to make everything a list and then grab it from the list